src/services/consensus/ConsensusDataService.ts

```diff
@@ -0,0 +1,67 @@
+import {inject, injectable} from 'inversify';
+import {Logger} from 'winston';
+import {LeavesAndFeeds} from '../../types/Consensus';
+import {FeedRepository} from '../../repositories/FeedRepository';
+import Leaf from '../../types/Leaf';
+import {FeedDataLoader} from './FeedDataLoader';
+import Feeds, {Feed} from '../../types/Feed';
+import {ConsensusDataGenerator} from './ConsensusDataGenerator';
+import dayjs from 'dayjs';
+import Settings from '../../types/Settings';
+
+@injectable()
+export class ConsensusDataService {
+  @inject('Logger') logger!: Logger;
+  @inject('Settings') settings!: Settings;
+  @inject(FeedRepository) feedRepository!: FeedRepository;
+  @inject(FeedDataLoader) feedDataLoader!: FeedDataLoader;
+  @inject(ConsensusDataGenerator) consensusDataGenerator!: ConsensusDataGenerator;
+
+  async getLeavesAndFeeds(timestamp: number): Promise<LeavesAndFeeds> {
+    const [startsAt, endsAt] = this.getTimeframe(timestamp);
+    const [fcdFeeds, leafFeeds] = await this.getActiveFeedMaps();
+    const feeds = this.joinFeedMaps([fcdFeeds, leafFeeds]);
+    const data = await this.feedDataLoader.apply({feeds, startsAt, endsAt});
+    console.log('[ConsesusDataService] data: ', JSON.stringify(data));
+    console.log('[ConsesusDataService] feeds: ', JSON.stringify(feeds));
+    const consensusData = await this.consensusDataGenerator.apply({feeds, data});
+    const firstClassLeaves = this.filterLeaves(consensusData, fcdFeeds);
+    console.log('[ConsesusDataService] firstClassLeaves: ', JSON.stringify(firstClassLeaves));
+    const leaves = this.filterLeaves(consensusData, leafFeeds);
+    console.log('[ConsesusDataService] consensusData: ', JSON.stringify(consensusData));
+    console.log('[ConsesusDataService] leafFeeds: ', JSON.stringify(leafFeeds));
+    console.log('[ConsesusDataService] leaves: ', JSON.stringify(leaves));
+    return {firstClassLeaves, leaves, fcdsFeeds: fcdFeeds, leavesFeeds: leafFeeds};
+  }
+
+  private getTimeframe(timestamp: number): Date[] {
+    return [
+      dayjs.unix(timestamp).subtract(this.settings.consensus.mintInterval, 'millisecond').toDate(),
```

Comment:
this is some new way of defining data range, Im not sure if this is good way. How we are doing it now for WS prices that are stored in redis?

Reply:
in other place we have `freshness`, so I believe this is wrong - we can not have two different settings for same feature. We need to have same timeframe across the app. Let's apply same solution that we have currently for WS prices. if we need it in few places let's create small service.

Reply:
Hey, I removed it and move to FeedDataLoader, please LMK if it is ok